### ISSUE TITLE
New features sugestion

### ISSUE THREAD

**Issue by dummy0stud** [DarthTon/Blackbone#48]
Hey, congratulations that's a very nice lib, but I have a couple of questions:

1 - Will inline hooking be available for remote hooking?
2 - Is it possible to hook functions inside x64 process from x86 one(using heaven's gate)? will this feature be implemented in a near future?

**Comment by DarthTon** [comment]
1. I'm working on hook injection. It will available as one of the hooking methods.
2. No. x86 process can't catch debug events of x64 process properly. 
3. There is a project configuration for XP.

**Comment by dummy0stud** [comment]
Will it work with some kind of IPC? If so, which IPC lib are you planning to use?

**Comment by DarthTon** [comment]
No. It won't do any IPC. You just write a hook and then inject it. After that it operates only inside target process.

**Comment by topkaly89** [comment]
@DarthTon How I can contact with you
Can you give me your email address?

**Comment by aeo24** [comment]
Any chance of SSDT hooks in the future?

**Comment by DarthTon** [comment]
I can implement some basic functionality. But the whole driver is designed to be PatchGuard compatible, whereas SSDT hooking is not. Also as you already said it's pretty easy for others to add hooking using existing stuff.

**Comment by aeo24** [comment]
Yes, just pure convenience for me. I looked into it but don't have everything set up for driver development. If you were to add an example that was commented/disabled that would be great starting point. Either way thanks for taking the time to consider. Great Project.

**Comment by dummy0stud** [comment]
obregistercallbacks is not a really good solution for protecting hwbps, you should use hypervisor to do the work

**Comment by chico-depressive** [comment]
@aeo24 there is a paper on phrack showing how to use hwbps to protect itself

**Comment by aeo24** [comment]
@chico-depressive Do you have a link?

**Comment by chico-depressive** [comment]
http://phrack.org/issues/65/8.html
